src/gui/share.cpp

```diff
@@ -0,0 +1,268 @@
+#include "share.h"
+#include "ocssharejob.h"
+#include "account.h"
+
+#include <QUrl>
+
+namespace OCC {
+
+Share::Share(AccountPtr account, const QString& id, const QString& path, int shareType,
+             int permissions, QObject *parent)
+: QObject(parent),
+  _account(account),
+  _id(id),
+  _path(path),
+  _shareType(shareType),
+  _permissions(permissions)
+{
+
+}
+
+const QString Share::getId()
+{
+    return _id;
+}
+
+int Share::getShareType()
+{
+    return _shareType;
+}
+
+int Share::getPermissions()
+{
+    return _permissions;
+}
+
+void Share::deleteShare()
+{
+    OcsShareJob *job = new OcsShareJob(_account, this);
+    connect(job, SIGNAL(jobFinished(QVariantMap)), this, SLOT(slotDeleted(QVariantMap)));
+    job->deleteShare(getId());
+}
+
+void Share::slotDeleted(const QVariantMap &reply)
+{
+    QString message;
+    int code = OcsShareJob::getJsonReturnCode(reply, message);
+    if (code != 100) {
+        //emit error!
+    }
+
+    emit shareDeleted();
```

Comment:
I don't remember: Does the OcsShareJob delete itself with deleteLater once done? If not, please check whether these jobs are cleaned up when done.